tests/unittests/test_molecule.py

```diff
@@ -57,9 +57,7 @@ def test_as_dict(self):
         """Test as dict"""
         as_dict = Molecule().as_dict()
```

Comment:
This looks very strange to me (all pytest code does, to be fair). I think it is an accident that `Molecule()` can be initiated without arguments. But a useful accident for testing, it seems!

Reply:
Well, it isn't really needed for this test though. But nice catch, I think you are right. We should almost certainly make the `name` argument to `Molecule` mandatory, what do you think?

Reply:
I've made it mandatory